src/utils/finance/source.py

```diff
@@ -0,0 +1,266 @@
+        """Add new columns that are copies of existing columns"""
+        for base_column, duplicate_column_list in self.duplicate_columns.items():
+            for duplicate_column in duplicate_column_list:
+                standard_data_table[duplicate_column] = standard_data_table[base_column]
+        return standard_data_table
+
+    def _add_new_columns(self, standard_data_table: pd.DataFrame) -> pd.DataFrame:
+        """Add new blank columns TODO: is this ever really needed?"""
+        for column in self.new_empty_columns:
+            standard_data_table[column] = None
+        return standard_data_table
+
+    def _add_state_code(self, standard_data_table: pd.DataFrame) -> pd.DataFrame:
+        """Add state code to relevent columns"""
+        for column in self.state_code_columns:
+            standard_data_table[column] = self.state_code
+        return standard_data_table
+
+    def standardize_schema(self, raw_data_table: pd.DataFrame) -> pd.DataFrame:
+        """Rename columns, remove unused columns, and add missing columns
+
+        Args:
+            raw_data_table: single dataframe with raw column names
+        """
+        standard_column_name_table = self._rename_columns(raw_data_table)
+        standard_relevant_column_table = self._drop_unused_columns(
+            standard_column_name_table
+        )
+        standard_relevant_column_table = self._add_state_code(
+            standard_relevant_column_table
+        )
+        standard_relevant_column_table = self._add_new_columns(
+            standard_relevant_column_table
+        )
+        standard_schema_table = self._add_duplicate_columns(
+            standard_relevant_column_table
+        )
+
+        return standard_schema_table
+
+
+class DataStandardizer:
+    """Ensures data conforms to expected types"""
+
+    def __init__(self, config_handler: ConfigHandler | None = None) -> None:
+        """Create a new DataStandardizer object
+
+        Args:
+            config_handler: ConfigHandler, documented # TODO
+        """
+        self.enum_mapper = config_handler.enum_mapper
+        self.column_to_date_format = config_handler.column_to_date_format
+
+    def _standardize_enums(self, standard_schema_table: pd.DataFrame) -> pd.DataFrame:
+        """Rename entity type columns"""
+        for column_name, column_enum_map in self.enum_mapper.items():
+            if column_name not in standard_schema_table.columns:
+                raise ValueError(f"Provided enum: {column_name} not in table")
+            # map each value in the table's column according to the
+            # provided enum mapper
+            standard_schema_table[column_name] = standard_schema_table[column_name].map(
+                column_enum_map
+            )
+        return standard_schema_table
+
+    def _standardize_column_to_date_format(
+        self, standard_schema_table: pd.DataFrame
+    ) -> pd.DataFrame:
+        for date_column in self.column_to_date_format:
```

Comment:
super nit: I prefer using `items()` here, since you need the date_format as well

Reply:
That's a good nit!